galsim/phase_psf.py

```diff
@@ -1085,6 +1094,14 @@ def _finalize(self, flux, suppress_warning):
         self.img *= (flux / (self.img.sum() * self.scale**2))
         self.img = galsim.ImageD(self.img.astype(np.float64), scale=self.scale)
 
+        if self.max_size is not None and self.max_size < self.img.scale*self.img.array.shape[0]:
+            center = self.img.center()
+            bds = galsim.BoundsI(center)
+            half_max_size = int(np.ceil(0.5*self.max_size/self.img.scale))
+            bds += center + galsim.PositionI(-half_max_size, -half_max_size)
+            bds += center + galsim.PositionI(half_max_size, half_max_size)
+            self.img = self.img[bds]
```

Comment:
I'm confused.  I thought the point was to limit the size of the InterpolatedImage so that future convolutions would potentially be faster.  If you limit the size earlier so that smaller FFT's are done as part of the Fourier optics (as is done in the older version of OpticalPSF) then don't you increase the aliasing?  I think we should avoid that.  I'm okay with just removing this keyword if you think this is a negligible speedup.